File: mediapicker/src/main/java/vn/tungdx/mediapicker/activities/MediaPickerErrorDialog.java
package vn.tungdx.mediapicker.activities;

import android.app.AlertDialog;
import android.app.Dialog;
import android.content.DialogInterface.OnClickListener;
import android.os.Bundle;
import android.support.v4.app.DialogFragment;

import vn.tungdx.mediapicker.R;

/**
 * @author TUNGDX
 */

/**
 * Create dialog for media picker module. Should only use in this module.
 */
public class MediaPickerErrorDialog extends DialogFragment {
    private String mMessage;
    private OnClickListener mOnPositionClickListener;

    public static MediaPickerErrorDialog newInstance(String msg) {
        MediaPickerErrorDialog dialog = new MediaPickerErrorDialog();
        Bundle bundle = new Bundle();
        bundle.putString("msg", msg);
        dialog.setArguments(bundle);
        return dialog;
    }

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        mMessage = getArguments().getString("msg");
    }

    @Override
    public Dialog onCreateDialog(Bundle savedInstanceState) {
        return new AlertDialog.Builder(getActivity())
                .setMessage(mMessage)
                .setPositiveButton(R.string.ok, mOnPositionClickListener)
                .create();
    }

    public void setOnOKClickListener(OnClickListener mOnClickListener) {
        this.mOnPositionClickListener = mOnClickListener;
    }
}

File: mediapicker/src/main/java/vn/tungdx/mediapicker/activities/MediaPickerErrorDialog.kt
package vn.tungdx.mediapicker.activities

import android.app.AlertDialog
import android.app.Dialog
import android.content.DialogInterface.OnClickListener
import android.os.Bundle
import androidx.fragment.app.DialogFragment

import vn.tungdx.mediapicker.R

/**
 * @author TUNGDX
 */

/**
 * Create dialog for media picker module. Should only use in this module.
 */
class MediaPickerErrorDialog : DialogFragment() {
    private var mMessage: String? = null
    private var mOnPositionClickListener: OnClickListener? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        mMessage = arguments!!.getString("msg")
    }

    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
        return AlertDialog.Builder(activity)
                .setMessage(mMessage)
                .setPositiveButton(R.string.ok, mOnPositionClickListener)
                .create()
    }

    fun setOnOKClickListener(mOnClickListener: OnClickListener) {
        this.mOnPositionClickListener = mOnClickListener
    }

    companion object {

        fun newInstance(msg: String): MediaPickerErrorDialog {
            val dialog = MediaPickerErrorDialog()
            val bundle = Bundle()
            bundle.putString("msg", msg)
            dialog.arguments = bundle
            return dialog
        }
    }
}